python_modules/dagster/dagster_tests/components_tests/state_backed_component_tests/test_state_backed_component.py

```diff
@@ -91,3 +102,43 @@ def test_simple_state_backed_component() -> None:
             spec = specs[0]
             new_metadata_value = spec.metadata["state_value"]
             assert new_metadata_value != original_metadata_value
+
+
+def test_multiple_components() -> None:
+    with instance_for_test(), create_defs_folder_sandbox() as sandbox:
+        sandbox.scaffold_component(
+            component_cls=MyStateBackedComponent,
+            defs_yaml_contents={
+                "type": "dagster_tests.components_tests.state_backed_component_tests.test_state_backed_component.MyStateBackedComponent",
+            },
+            defs_path="first",
+        )
+
+        second_component_path = sandbox.scaffold_component(
+            component_cls=MyStateBackedComponent,
+            defs_yaml_contents={
+                "type": "dagster_tests.components_tests.state_backed_component_tests.test_state_backed_component.MyStateBackedComponent",
+            },
+            defs_path="second",
+        )
+
+        with pytest.raises(
+            ComponentTreeException,
+            match="MyStateBackedComponent",
```

Comment:
match should include `get_defs_state_key` to ensure we are pointing to way to solve problem